models/utils_test.go

```diff
@@ -0,0 +1,207 @@
+			"int16":   int16(0),
+			"int32":   int32(0),
+			"int64":   int64(0),
+			"uint":    uint(0),
+			"uint8":   uint8(0),
+			"uint16":  uint16(0),
+			"uint32":  uint32(0),
+			"uint64":  uint64(0),
+			"pointer": pointer,
+			"nil":     nil,
+		}
+
+		require.NotEmpty(t, data)
+		models.StripEmptyEntries(data)
+		require.Empty(t, data)
+	})
+
+	t.Run("should not omit non-empty basic types", func(t *testing.T) {
+		data := map[string]any{
+			"array":   [1]any{1},
+			"map":     map[any]any{"foo": "bar"},
+			"slice":   []any{1},
+			"string":  "foo",
+			"bool":    true,
+			"float32": float32(1),
+			"float64": float64(1),
+			"int":     int(1),
+			"int8":    int8(1),
+			"int16":   int16(1),
+			"int32":   int32(1),
+			"int64":   int64(1),
+			"uint":    uint(1),
+			"uint8":   uint8(1),
+			"uint16":  uint16(1),
+			"uint32":  uint32(1),
+			"uint64":  uint64(1),
+			"pointer": &Foo{},
+		}
+
+		require.NotEmpty(t, data)
+		numKeys := len(data)
+		models.StripEmptyEntries(data)
+		require.NotEmpty(t, data)
+		require.Equal(t, numKeys, len(data))
+	})
+
+	t.Run("should not omit empty struct types", func(t *testing.T) {
+		data := map[string]any{
+			"struct": Foo{Bar: "baz"},
+		}
+
+		require.NotEmpty(t, data)
+		numKeys := len(data)
+		models.StripEmptyEntries(data)
+		require.NotEmpty(t, data)
+		require.Equal(t, numKeys, len(data))
+		require.NotEmpty(t, data["struct"])
+		require.Equal(t, data["struct"].(Foo).Bar, "baz")
+	})
+
+	t.Run("should recursively strip non-empty, nested map[string]any entries", func(t *testing.T) {
+		data := map[string]any{
+			"empty": map[string]any{},
+			"nonempty": map[string]any{
+				"emptyprop":    0,
+				"nonemptyprop": 1,
+			},
+		}
+
+		models.StripEmptyEntries(data)
+		require.Nil(t, data["empty"])
+		require.NotNil(t, data["nonempty"])
+		require.IsType(t, map[string]any{}, data["nonempty"])
+		nested := data["nonempty"].(map[string]any)
+		require.Equal(t, 1, len(nested))
+		require.Nil(t, nested["emptyprop"])
+		require.Equal(t, 1, nested["nonemptyprop"])
+	})
+
+	t.Run("should strip non-empty slice entries of type map[string]any", func(t *testing.T) {
+		data := map[string]any{
+			"empty": []any{},
+			"nonempty": []any{
+				map[string]any{
+					"emptyprop":    0,
+					"nonemptyprop": 1,
+				},
+			},
+		}
+
+		fmt.Println(data["nonempty"])
+		models.StripEmptyEntries(data)
+		require.Nil(t, data["empty"])
+		require.NotNil(t, data["nonempty"])
+		require.IsType(t, []any{}, data["nonempty"])
+		slice := data["nonempty"].([]any)
+		fmt.Println(slice)
+		require.IsType(t, map[string]any{}, slice[0])
+		entry := slice[0].(map[string]any)
+		require.Nil(t, entry["emptyprop"])
+		require.Equal(t, 1, entry["nonemptyprop"])
+	})
+}
+
+func TestOmitEmpty(t *testing.T) {
+	t.Run("should omit empty basic types", func(t *testing.T) {
+		data := json.RawMessage(`{
+			"string":   "",
+			"number":   0,
+      "object": {},
+      "array": [],
+      "boolean": false,
+      "null": null
```

Comment:
Looks like these somehow got set as space indented while the others are tab indented. Not a big deal, but might be worth formatting to prevent mixed tab/space